mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/model/QuotaThresholds.java

```diff
@@ -0,0 +1,51 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.quota.model;
+
+import java.util.Comparator;
+import java.util.List;
+
+import org.apache.james.mailbox.model.Quota;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+
+public class QuotaThresholds {
+    private final ImmutableList<QuotaThreshold> quotaThresholds;
+
+    public QuotaThresholds(List<QuotaThreshold> quotaThresholds) {
+        this.quotaThresholds = ImmutableList.copyOf(quotaThresholds);
+    }
+
+    public QuotaThreshold firstExceededThreshold(Quota quota) {
+        return quotaThresholds.stream()
+            .sorted(Comparator.comparing(QuotaThreshold::getQuotaOccupationRatio).reversed())
+            .filter(quotaLevel -> quotaLevel.isExceeded(quota))
+            .findFirst()
+            .orElse(QuotaThreshold.ZERO);
```

Comment:
What is this special value ? It looks like a "Null Object" Pattern, but it's not clear to me

Reply:
yes, I guess the function should return an Optional, no?

Reply:
I just found that relying on an implicit Zero threshold simplified things a lot.

Reply:
Null Object is way better than Optional, and after thinking about it, Zero is probably the right name.